server/src/main/java/org/elasticsearch/transport/SniffConnectionStrategy.java

```diff
@@ -508,6 +525,14 @@ private static boolean seedsChanged(final List<String> oldSeedNodes, final List<
         return oldSeeds.equals(newSeeds) == false;
     }
 
+    private static boolean authorizationChanged(String oldOptionalCredential, String newOptionalCredential) {
+        if (oldOptionalCredential == null || oldOptionalCredential.isEmpty()) {
+            return (newOptionalCredential == null || newOptionalCredential.isEmpty()) == false;
+        }
+
+        return Objects.equals(oldOptionalCredential, newOptionalCredential) == false;
```

Comment:
Can simplify to:

```
return Strings.isNullOrEmpty(oldOptionalCredential) != Strings.isNullOrEmpty(newOptionalCredential) || Objects.equals(oldOptionalCredential, newOptionalCredential) == false;
```

Reply:
My intention was to copy `proxyChanged` and replace _proxy_ with _authorization_. If you want me to modify `authorizationChanged`, do you think it is worthwhile applying the same pattern to `proxyChanged`?

https://github.com/elastic/elasticsearch/blob/b1eecb712f5126c297bba20b08dd112aa69698d2/server/src/main/java/org/elasticsearch/transport/SniffConnectionStrategy.java#L528-L534

https://github.com/elastic/elasticsearch/blob/b1eecb712f5126c297bba20b08dd112aa69698d2/server/src/main/java/org/elasticsearch/transport/SniffConnectionStrategy.java#L536-L542

Reply:
I would not mind either way.